### PR TITLE
Adding code changes since `notice of a revert` occurred

### PR DESCRIPTION
**Pull request by pswish**
Adding some changes from last few days. Please review and comment/approve when you have a chance!

### CODE REVIEW

File: dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java [hunk 1]
```diff
@@ -44,8 +45,10 @@ public Iterator iterator() {
 
     @Override
     public Object[] toArray() {
-        // Object[] objects = l1.toArray(new String[0]); needs new logic
-        return new Object[0];
+        List<T> al1= new ArrayList<T>();
+        this.add(al1);
+        al1.toArray();
```
Comment (WestDav) [inline dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java:50]: this needs to be changed

Comment (nathanMcL) [reply dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java:50]: look forward to compiling.

**Review by nathanMcL — APPROVED** [review]

**Review by EvergreenTor — APPROVED** [review]
like it!

File: dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java (hunk 1 above)
Comment (pswish) [reply dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java:50]: > this needs to be changed

Good idea, will work to add it in the next pull request

**Pull request merged**